runtime/cudaq/builder/kernel_builder.h

```diff
@@ -310,6 +310,14 @@ class kernel_builder_base {
   virtual std::string to_quake() const = 0;
   virtual void jitCode(std::vector<std::string> extraLibPaths = {}) = 0;
   virtual ~kernel_builder_base() = default;
+
+  /// @brief Write the kernel_builder to the given output stream.
+  /// This outputs the Quake representation.
+  friend std::ostream &operator<<(std::ostream &stream,
```

Comment:
This may need an attribute to make sure the linker doesn't throw it away. Not sure.